lib/rust/ensogl/core/src/timer/interval.rs

```diff
@@ -0,0 +1,84 @@
+//! FRP bindings for browser `setInterval` timer.
+
+use crate::prelude::*;
+
+use enso_web::traits::ClosureOps;
+use enso_web::window;
+use enso_web::Closure;
+
+use std::time::Duration;
+
+/// Closure type alias for use in `setInterval` call.
+type TimerClosure = Closure<dyn FnMut()>;
+
+/// A repating timer based on `setIntevral` browser API.
+///
+/// The timer can be started or stopped at any time. The timer value
+/// is provided during the start.
+///
+/// When the timer expires, `expired` is emitted.
+#[derive(Clone, CloneRef, Debug)]
+pub struct Interval {
+    /// Starts the timer immediately with provided interval value.
+    /// If the timer was already started, it is stopled and restarted.
+    pub start:       frp::Any<Duration>,
+    /// Stops the timer if it is running. No `on_interval` events
+    /// will be emitted until it is started again.
+    pub stop:        frp::Any,
+    /// Emitted when timer interval is reached. Events will be emitted
+    /// repeatedly until the timer is stopped.
+    pub on_interval: frp::Stream<()>,
+    raw_interval:    Rc<RawInterval>,
+}
+
+impl Interval {
+    /// Constructor. Timer is initially not started.
+    pub fn new(network: &frp::Network) -> Self {
+        frp::extend! { network
+            on_interval <- any_mut();
+        }
+
+        let closure: TimerClosure = Closure::new(f!([on_interval]() on_interval.emit(())));
+        let raw_interval = Rc::new(RawInterval::new(closure));
+
+        frp::extend! { network
+            start <- any_mut::<Duration>();
+            stop  <- any_mut::<()>();
+            eval  start((duration) raw_interval.start(duration.as_millis() as i32));
+            eval_ stop(raw_interval.stop());
```

Comment:
This implementation creates new FRP nodes in the network passed as argument. Is this the desired behavior? Why is it better than this struct owning the network in order to minimise possibility of mem leak? Also, if we want to keep this design, there is no warning about it in the docs.